src/Domain/FunctionsOfTime/PiecewisePolynomial.cpp

```diff
@@ -47,7 +47,7 @@ PiecewisePolynomial<MaxDeriv>& PiecewisePolynomial<MaxDeriv>::operator=(
     return *this;
   }
   deriv_info_at_update_times_ = rhs.deriv_info_at_update_times_;
-  expiration_time_ = rhs.expiration_time_;
+  expiration_time_.store(rhs.expiration_time_.load(std::memory_order_relaxed));
```

Comment:
It looks a bit weird to be specifying memory orders for the loads but not the stores.  Is this intentional?  (Happens in a bunch of places.)

Reply:
I believe since we don't expect to copy the classes, any ordering should be fine. That's not to disagree with your request of consistency. The default is the strongest, but should be okay unless we copy this class a lot (I hope not)